Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/transform/AuditTransform.java

```diff
@@ -0,0 +1,579 @@
+        EventIdentificationType oEventIdentificationType = createEventIdentification(
+            isRequesting ? getEventActionCodeRequestor(serviceName) : getEventActionCodeResponder(serviceName),
+            AuditTransformConstants.EVENT_OUTCOME_INDICATOR_SUCCESS, eventID);
+        return oEventIdentificationType;
+    }
+
+    /**
+     *
+     * @param isRequesting
+     * @param webContextProperties
+     * @param serviceName
+     * @return
+     */
+    protected AuditMessageType.ActiveParticipant getActiveParticipantSource(boolean isRequesting, Properties webContextProperties, String serviceName) {
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        participant.setUserID(AuditTransformConstants.ACTIVE_PARTICPANT_USER_ID_SOURCE);
+        participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        String hostAddress = null;
+        hostAddress = isRequesting ? getLocalHostAddress() : getRemoteHostAddress(webContextProperties);
+        participant.setNetworkAccessPointID(hostAddress);
+        participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(hostAddress));
+
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(AuditTransformConstants.ACTIVE_PARTICIPANT_ROLE_CODE_Source, null,
+            AuditTransformConstants.ACTIVE_PARTICIPANT_CODE_SYSTEM_NAME, AuditTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+        participant.setUserIsRequestor(isRequesting);
+        return participant;
+    }
+
+    /**
+     *
+     * @param target
+     * @param isRequesting
+     * @param webContextProprties
+     * @param serviceName
+     * @return
+     */
+    protected AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target, boolean isRequesting, Properties webContextProprties, String serviceName) {
+        String strUrl = null;
+        String strHost = null;
+        boolean setDefaultValue = true;
+
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        strUrl = isRequesting ? getWebServiceUrlFromRemoteObject(target, serviceName) : getWebServiceRequestURL(webContextProprties);
+        if (strUrl != null) {
+            try {
+                URL url = new URL(strUrl);
+                participant.setUserID(strUrl);
+                strHost = url.getHost();
+                participant.setNetworkAccessPointID(strHost);
+                participant.setNetworkAccessPointTypeCode(getNetworkAccessPointTypeCode(strHost));
+                setDefaultValue = false;
+            } catch (MalformedURLException ex) {
+                LOG.error(ex);
+                setDefaultValue = true;
+            }
+        }
+        //if the url is null or not a valid url
+        if (setDefaultValue) {
```

Comment:
Can this logic be added in the catch, this eliminating the need for both this "if" statement as well as the "setDefaultValue" variable?